src/manager/InMemoryTaskManager.java

```
@@ -42,86 +54,186 @@ public int getIdCounter() {
         return id;
     }
 
-
     private int generatedId() {
         return ++id;
     }
 
     @Override
     public Task getTask(int id) {
-        inMemoryHistoryManager.add(tasks.get(id));
-        return tasks.get(id);
+        if (tasks.containsKey(id)) {
```

Лучше стараться чтобы код был ближе к левой стороне. Если можно обойтись без сдвигания его вправо, то лучше этого не делать. В данной ситуации можно if (!tasks.containsKey(id)) {throw new NotFoundException();}
А дальше уже код метод без сдига вправо. Потому что когда кода станет больше он будет сильно уходить вправо и читать его станет сложнее. Плюс в данной ситуации мы применили подход fail fast - как можно раньше выходить из метода, если входные аргументы не удовлетворяют требованиям.
Ниже в методах можно тоже применять аналогичные подходы.